src/game_commonevent.cpp

```diff
@@ -109,3 +88,17 @@ RPG::SaveEventData Game_CommonEvent::GetSaveData() {
 
 	return event_data;
 }
+
+bool Game_CommonEvent::IsWaitingForegroundExecution() const {
+	auto* ce = ReaderUtil::GetElement(Data::commonevents, common_event_id);
+	return ce->trigger == RPG::EventPage::Trigger_auto_start &&
+		(!ce->switch_flag || Game_Switches.Get(ce->switch_id))
+		&& !ce->event_commands.empty();
+}
+
+bool Game_CommonEvent::IsWaitingBackgroundExecution() const {
+	auto* ce = ReaderUtil::GetElement(Data::commonevents, common_event_id);
+	return ce->trigger == RPG::EventPage::Trigger_parallel &&
+		(!ce->switch_flag || Game_Switches.Get(ce->switch_id))
+		&& !ce->event_commands.empty();
+}
```

Comment:
Maybe these could be combined?

Reply:
Sure, I'll drop a commit for that into the next one #1687